roles/postgresql-function/tasks/main.yml

```diff
@@ -0,0 +1,14 @@
+---
+
+- name: Copy function file
```

Comment:
copying the function texts as a file seems redundant 

it is enough to execute this code via psql

Reply:
Agreed. That was my initial thought also but wasn't quite sure how to do it inline (escaping the $$ etc.)

Reply:
Can you give me a hand with this?

Reply:
```yaml
- name: "Create function 'user_search' for pgbouncer 'auth_query' option in all databases"
  become: yes
  become_user: postgres
  ansible.builtin.shell: |
    for db in $({{ postgresql_bin_dir }}/psql -p {{ postgresql_port }} -U {{ patroni_superuser_username }} -tAXc \
    "select datname from pg_catalog.pg_database where datname <> 'template0'"); do
      {{ postgresql_bin_dir }}/psql -p {{ postgresql_port }} -U {{ patroni_superuser_username }}  -d "$db" -tAXc '
        CREATE OR REPLACE FUNCTION user_search(uname TEXT) RETURNS TABLE (usename name, passwd text) AS
        $$
        SELECT usename, passwd FROM pg_shadow WHERE usename=$1;
        $$
        LANGUAGE sql SECURITY DEFINER;
      '; done
  args:
    executable: /bin/bash
  tags: pgbouncer, pgbouncer_conf, pgbouncer_auth_query
```